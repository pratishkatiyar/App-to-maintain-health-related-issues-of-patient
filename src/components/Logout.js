import React from 'react'
import { Link } from 'react-router-dom'
import { useLocation } from "react-router-dom"
import { Redirect } from 'react-router-dom';
import Navbar from './NavBar';

const Logout = () => {
    const location = useLocation()
    const closexit = location.state?.closeExit;  
    return <div data-testid='logout-1'>
        <Navbar/>
        {closexit ? (localStorage.removeItem("token")):(<Redirect to="/"/>)}
        <h1>You have been logged out</h1>
        <Link to = '/'><button className="button-19">Login Again</button></Link>        
    </div>;    
};

export default Logout;

