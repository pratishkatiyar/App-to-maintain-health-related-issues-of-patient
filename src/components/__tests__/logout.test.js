import { render, screen, cleanup } from '@testing-library/react'
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Logout from '../Logout'
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

afterEach(() => {
    cleanup();
})

test('should render Logout component', () =>{
    render(<BrowserRouter><Logout/></BrowserRouter>);
    const logoutElement = screen.getByTestId('logout-1');
    expect(logoutElement).toBeInTheDocument();
    expect(logoutElement).toHaveTextContent('You have been logged out');
})

test('matches snapshot', ()=> {
    const tree = renderer.create(<BrowserRouter><Logout/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})