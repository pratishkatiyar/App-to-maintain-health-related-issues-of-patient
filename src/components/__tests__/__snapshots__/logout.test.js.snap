// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="logout-1"
>
  <div
    data-testid="navbar-1"
  >
    <nav>
      <div
        className="logo"
      >
        <p>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </p>
      </div>
      <div
        className="logo"
      >
        Patient Registration App
      </div>
      <div />
    </nav>
  </div>
  <h1>
    You have been logged out
  </h1>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="button-19"
    >
      Login Again
    </button>
  </a>
</div>
`;
